appserver/web/web-core/src/main/java/org/apache/catalina/session/PersistentManagerBase.java

```diff
@@ -728,14 +738,35 @@ public Session findSession(String id) throws IOException {
         //6406580 END        
 
         Session session = super.findSession(id);
-        if (session != null)
+
+        if (session != null) {
+            //verify if session also is on the store and compare lastAccessTime and thisAccessedTime
+            Session sessionFromStore = swapIn(id);
+            if (sessionFromStore != null) {
+                compareAndUpdateAccessedTime((StandardSession) session, (StandardSession) sessionFromStore);
+            }
             return (session);
+        }
 
         // See if the Session is in the Store
         session = swapIn(id);
         return (session);
 
-    }       
+    }
+
+    public void compareAndUpdateAccessedTime(StandardSession currentSession, StandardSession sessionFromStore) {
```

Comment:
Any possibilities of `sessionFromStore.getLastAccessedTimeInternal() == currentSession.getLastAccessedTimeInternal()
&& sessionFromStore.getThisAccessedTime() > currentSession.getThisAccessedTime()` ?

Reply:
I changed the implementation because of the new functionality to have a new thread that validates the last values added on the storage